Collection view

ThanyaniDev opened the pull request:
I have added an updated collection view screen
![Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-09 at 07 41 23](https://user-images.githubusercontent.com/55865505/98504173-f9506d80-225e-11eb-8138-6e8b3b5ae5b6.png)
And I have also added functions that convert UTC to Local Date
<img width="532" alt="Screenshot 2020-11-09 at 07 35 43" src="https://user-images.githubusercontent.com/55865505/98504223-200ea400-225f-11eb-9d6e-44f06dd424e0.png">

Commit:
added convert UTC to localDate

Commit:
added collection view

Sashen943 commented on MarsMission/Controllers/LandingViewController.swift:
```diff
@@ -10,62 +10,60 @@ import UIKit
-		cell.backgroundColor = .blue
-		cell.temp?.text = String(forecastWeatherData[indexPath.row].forecasts[indexPath.row].temp!)
+		cell.backgroundColor = .darkGray
+		cell.dateLabel.text = "weather"
```
Remove these hard coded values

Sashen943 commented on MarsMission/Controllers/LandingViewController.swift:
```diff
@@ -10,62 +10,60 @@ import UIKit
+
 	func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-		return forecastWeatherData.count
+		return 6
```
Should return the size of the array. Remove this hard coded 6

Sashen943 commented on MarsMission/Models/WeatherData.swift:
```diff
@@ -8,19 +8,15 @@
 
 struct WeatherData: Codable {
-	var forecasts: [forecastsData]
+	var forecasts: [forecast]
```
class Forecast should be named with a capital. Remember that all classes in swift are named with a capital

Sashen943 commented on MarsMission/Models/WeatherData.swift:
```diff
@@ -8,19 +8,15 @@
 }
 
-struct forecastsData: Codable {
+struct forecast: Codable {
```
Capitalise class name

Sashen943 commented on MarsMission/ViewModels/WeatherDataViewModel.swift:
```diff
@@ -35,7 +36,10 @@ class WeatherDataViewModel {
 			self.weatherData = weatherData
 			self.view.populateWeatherData(weatherData.lastUpdated!,
-										  weatherData.weatherStation!)
+										  weatherData.weatherStation!,
```
Why do you have this function if you're using a collection view? Make sure to understand how arrays work

Sashen943 commented on MarsMission/Views/WeatherView.swift:
```diff
@@ -8,7 +8,6 @@
 
 protocol WeatherView {
-	func populateWeatherData(_ lastUpdated: String,_ weatherStation: String)
+	func populateWeatherData(_ lastUpdated: String,_ weatherStation: String,_ date: String,_ temp: Double)
```
Why do you have this function if you're using a collection view? Please ensure that you understand how the UI Components that you are using works before you use them

Sashen943 commented on MarsMission/Helper/UTCToDate.swift:
```diff
@@ -7,15 +7,12 @@
+	let convertedDate =  format.date(from: date)
+	format.timeZone =  TimeZone.current
+	format.dateFormat = "MMM d, h:mm a"
+	let localDateString =  format.string(from: convertedDate!)
```
Remove forceful unwrapping here

Sashen943 commented on MarsMission/Controllers/LandingViewController.swift:
```diff
@@ -10,62 +10,60 @@ import UIKit
 
 extension LandingViewController: WeatherView {
-	func setUpCollectionView() {
-		let nib = UINib(nibName: "weatherForecastViewCell", bundle: nil)
```
Why is you nib named with a lower case?

Sashen943 reviewed (changes requested):
Please book some time with me to understand how collection views and arrays work

Commit:
project refactoring

ThanyaniDev commented:
<img width="324" alt="Screenshot 2020-11-11 at 05 50 18" src="https://user-images.githubusercontent.com/55865505/98764397-63dee600-23e4-11eb-93e2-ad75197df025.png">
<img width="584" alt="Screenshot 2020-11-11 at 01 56 51" src="https://user-images.githubusercontent.com/55865505/98764439-73f6c580-23e4-11eb-863c-e3d603d44764.png">
Added Project Structure Changes and added nil coalescing

ThanyaniDev commented:
<img width="331" alt="Screenshot 2020-11-11 at 07 48 27" src="https://user-images.githubusercontent.com/55865505/98773815-65afa600-23f2-11eb-8783-2b24d3002ff0.png">

<img width="584" alt="Screenshot 2020-11-11 at 01 56 51" src="https://user-images.githubusercontent.com/55865505/98773852-74965880-23f2-11eb-8520-adafca5f4613.png">

Commit:
added a structure update and  nil coalescing  on UTCtoDate function

Commit:
added ActivityIndicator

Commit:
Detailed view added.

Commit:
passing data through navigation controller

Commit:
refactoring forecastDetailViewController function

ThanyaniDev commented:
![Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-12 at 03 50 53](https://user-images.githubusercontent.com/55865505/98885233-571cc980-249a-11eb-8eae-a132daf083dd.png)
Updated forecast View.
![Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-12 at 03 51 00](https://user-images.githubusercontent.com/55865505/98885324-7d426980-249a-11eb-9ae7-286b36ace2bc.png)
forecast detail view

Commit:
added a String extension to simplfy string entry in functions

Commit:
time conversion adding TimeZone and  Cleaning the codebase

Commit:
Added Dependency injection on ForecastRepositoryImplemantation  and a ternary operator on  ForecastDetailViewController for forecastDetailSafe
